Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -4236,6 +4236,9 @@ - (NSDictionary *)textStylingAtPosition:(UITextPosition *)position inDirection:(
 
     if (!_page->editorState().postLayoutData)
         return nil;
+
+    [result setObject:[UIColor blackColor] forKey:NSForegroundColorAttributeName];
```

Comment:
I wonder if we should always return `+blackColor` in the fallback paths above, when post-layout data isn't up to date? i.e.

```
- (NSDictionary *)textStylingAtPosition:(UITextPosition *)position inDirection:(UITextStorageDirection)direction
{
    NSMutableDictionary* result = [NSMutableDictionary dictionary];
    [result setObject:[UIColor blackColor] forKey:NSForegroundColorAttributeName];
    if (!position || !_page->editorState().isContentRichlyEditable)
        return result;

    if (!_page->editorState().postLayoutData)
        return result;
```